lib/vehicle_factory.rb

```diff
@@ -0,0 +1,95 @@
+class VehicleFactory
+  attr_reader :created_vehicles
+
+  def initialize
+    @created_vehicles = []
+  end
+
+  def create_vehicles(vehicles)
+    if vehicles[0][:electric_vehicle_type]
+      vehicles.each do |vehicle_data|
+        vehicle = Vehicle.new({
+          vin: vehicle_data[:vin_1_10],
+          year: vehicle_data[:model_year],
+          make: vehicle_data[:make],
+          model: vehicle_data[:model],
+          engine: :ev,
+          county: vehicle_data[:county]
+          })
+          @created_vehicles << vehicle
+      end
+      @created_vehicles
+    elsif vehicles[0][:record_type]
```

Comment:
Could potentially use some boolean logic to merge these two blocks together. The if and the elsif. Could do something like `if vehicles[0][:electric_vehicile_type] || vehicles[0][:record_type]` and then use that same logic when setting the keys `vin: vehicle_data[:vin_1_10] || vehicle_data[:vin]`